src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -896,8 +896,11 @@ public static void Pow10(uint exponent, out BigInteger result)
                     if ((exponent & 1) != 0)
                     {
                         // Multiply into the next temporary
-                        ref BigInteger rhs = ref *(BigInteger*)(Unsafe.AsPointer(ref s_Pow10BigNumTable[s_Pow10BigNumTableIndices[index]]));
-                        Multiply(ref lhs, ref rhs, ref product);
+                        fixed (uint* pBigNumEntry = &s_Pow10BigNumTable[s_Pow10BigNumTableIndices[index]])
```

Comment:
We could get rid of this pinning if/when https://github.com/dotnet/designs/pull/46 becomes supported (as the data would become an actual RVA static, rather than getting copied to a GC tracked array).

Reply:
This can't be:
```
ref BigInteger rhs = ref Unsafe.As<uint, BigInteger>(ref s_Pow10BigNumTable[s_Pow10BigNumTableIndices[index]]);
```
because BigInteger is a ref struct?  Is there a technical reason it needs to be a ref struct?

Reply:
Technically no. We rely on it only existing on the stack for various optimizations, however and a goal is to get rid of the unsafe (`fixed-sized buffer`) code in favor of it carrying around a `Span<T>` (much like we did with `Number.NumberBuffer`